src/thunderbots/software/ai/hl/stp/play/play.cpp

```diff
@@ -0,0 +1,28 @@
+#include "ai/hl/stp/play/play.h"
+
+const std::vector<std::shared_ptr<PlayFactory>>& Play::getRegistry()
+{
+    return Play::getMutableRegistry();
+}
+
+std::vector<std::shared_ptr<PlayFactory>>& Play::getMutableRegistry()
+{
+    static std::vector<std::shared_ptr<PlayFactory>> instance;
```

Comment:
I think this should this be a static class member instead?

Reply:
No, we want it in the function to guarantee it is initialized before it is accessed. The registry must be accessed through the `get` functions, and this guarantees the instance will be initialized before it is returned/accessed.

If the instance was a class member we can't guarantee it will be initialized before the Factories and other static objects that might access it. I tried it with a class member before and it didn't work.